github/VisualStudio
# Theme GitHub dialog window borders

The border color of dialog windows is dependent on the current Windows desktop background (unless the color is set explicitly). This might be inappropriate for the current Visual Studio theme, making the border invisible or look out of place. The only way to change this is to replace the dialog windows border, title and glyphs. The PR creates a style for GitHiub dialog windows based on the active Visual Studio theme colors.

### What this PR does

- Use `ThemedDialogColors.WindowBorderBrushKey` for border color
- Use the close glyph from other Visual Studio windows
- Don't show dialog title (this is consistent with other Visual Studio theme dialogs)
- Allow dialog window to be dragged by any non-active part
- Allow dialog window to be resized by grabbing any side
- Use same colors for search/password text box as standard text boxes
- Don't show dialog in task bar (consistent with other Visual Studio dialogs)
- Use `ThemedDialogColors` instead of `EnvironmentColors` (which were sometimes missing or wrong)

### Before and after

Before:
![image](https://user-images.githubusercontent.com/11719160/58011815-00277580-7aeb-11e9-857a-d3e65b39c9ef.png)

After:
![image](https://user-images.githubusercontent.com/11719160/58011854-0fa6be80-7aeb-11e9-9bdc-fcae74a03388.png)

### How to test

Click on `File > Clone or checkout code > GitHub`:

1. Check border color is appropriate for different Visual Studi0 themes (`Blue`, `Dark` and `Light`)
2. Check border color is visible in High contrast mode (use the Windows 10 `High contrast settings`)
3. Check search text box uses same colors as local path text box
4. Check the dialog can be moved around
5. Check the dialog can be resized around
6. Check dialog works without the window title (is what's going on obvious from its context)
7. Check no icon appears on task bar for dialog
8. Check the dialog can be closed

Fixes #2356
